lib/src/markdown.service.ts

```diff
@@ -390,15 +394,11 @@ export class MarkdownService {
     if (!isPlatformBrowser(this.platform)) {
       return;
     }
-    if (typeof mermaid === 'undefined' || typeof mermaid.init === 'undefined') {
+    if (typeof mermaid === 'undefined' || typeof mermaid.initialize === 'undefined') {
       throw new Error(errorMermaidNotLoaded);
     }
-    const mermaidElements = element.querySelectorAll('.mermaid');
```

Comment:
`element` now is unsued and should be removed

Reply:
Now that you bring that to my attention, the changes that has been submitted use the query selector parameter to the `mermaid.run` function which probably look in the whole document as opposed to the previous implementation that was using the nodes within the provided element... so it might end up rendering the same node multiple times if more than one markdown component/directive is used in the same page.

Il will fix it using the nodes parameter in the `mermaid.run` function.

Reply:
Done in the PR #477